Python/Product/Analysis/Infrastructure/ProcessHelper.cs

```diff
@@ -126,11 +138,12 @@ sealed class ProcessHelper : IDisposable {
                 throw new InvalidOperationException("Process was not started");
             }
 
-            await Task.WhenAll(
-                _seenNullOutput.WaitAsync(cancellationToken),
-                _seenNullError.WaitAsync(cancellationToken)
-            );
-            cancellationToken.ThrowIfCancellationRequested();
+            if (!_process.HasExited) {
+                await _seenNullOutput.WaitAsync(cancellationToken).ConfigureAwait(false);
+                await _seenNullError.WaitAsync(cancellationToken).ConfigureAwait(false);
+                cancellationToken.ThrowIfCancellationRequested();
```

Comment:
Do you really want to throw OCE when process has completed it's work?

Reply:
Semantically, the decision has been made to cancel, which presumably means we also want to cancel any subsequent work as well. Though since we've just completed a cross-process task without cancellation and are not marshalling back to the original thread here, it's very unlikely that a cancellation will occur between the previous wait and this check. (I suspect the call has stuck around from a previous iteration of this code... took a few goes to get `Process` termination right.)